src/ashell/comms-uart.c

```diff
@@ -322,31 +303,18 @@ void comms_write_buf(const char *buf, int len)
     if (len == 0)
         return;
 
-    struct device *dev = dev_upload;
-    uart_irq_tx_enable(dev);
-
+    uart_irq_tx_enable(dev_upload);
     data_transmitted = false;
-    uart_fifo_fill(dev, buf, len);
-
+    uart_fifo_fill(dev_upload, (const uint8_t *)buf, len);
```

Comment:
> The function @pfalcon pointed to shows that it will return 0 if there is no such api in the driver; so if that could ever happen we would hang in my above code.

Well, that's not the problem to care about. If CONFIG_UART_INTERRUPT_DRIVEN is defined, that function can't be non-empty, it's part of the core interrupt-driven API. The problem is like above - you can't expect it to write everything you asked it for.

Code like that (not checking return) is the underlying cause of all those "can't get lines longer 90 chars" and "it doesn't work on any other hardware" problems.